war/src/main/webapp/scripts/sortable.js

```diff
@@ -265,28 +265,43 @@ var Sortable = (function() {
     arrowTable.up.next = arrowTable.down;
     arrowTable.down.next = arrowTable.up;
 
-
+    /**
+     * Matches dates like:
+     *   "1/4/2017 ignored content"
+     *   "03-23-99 1:30 PM also ignored content"
+     *   "12-25/1979 13:45:22 always with the ignored content!"
+     */
+    var date_pattern = /^(\d{1,2})[\/-](\d{1,2})[\/-](\d\d|\d\d\d\d)(?:(?:\s*(\d{1,2})?:(\d\d)?(?::(\d\d)?)?)?(?:\s*([aA][mM]|[pP][mM])?))\b/
```

Comment:
Any support for ISO 8601 dates?

Reply:
@daniel-beck that's a good idea but I think it will add significant complexity. My intent was to improve the existing pattern matching but preserve the existing behavior, so I figured ISO 8601 was out of scope.

If I am to add ISO 8601 handling, I'd want to put in unit tests to guarantee I'm not adding any regressions, but I'm not typically a js coder so I'm not sure where to start with that. I'll look through the project today to see if I can find where other JavaScript code is getting tested, or a developer doc, but if you can point me in the right direction it would be a great help. 🙂

I do really like the idea though. Maybe it should go in as a separate pr?

Reply:
Hmm, I wrote that on my phone and didn't proofread it evidently. By "add significant complexity" I meant "add some small complexity which might have significant side-effects". It would be simple to implement but I'd be worried about changing behavior for plugins that depend on `sortable.js` (and actually have date columns that work with `sortable`, unlike the Build Failure Analyzer plugin which is what I'm trying to fix in the first place).

Reply:
The regex is not really readable (and so will be hard to maintain) but seems to work as expected.

- `1/4/2017` ignored content
- `03-23-99 1:30 PM` also ignored content
- `03-23-99 1:30       PM`      also ignored content
- `12-25/1979 13:45:22` ignored content
- `12-25/1979 13:45:` ignored content
- `12-25/1979 13:45` ignored content
- `12-25/1979 13:` ignored content
- `12-25/1979 1:` ignored content
- `12-25/1979 13:`0 ignored content (0 meaning what ? 0 minutes ? in that case we display 00 normally)
- `12-25/1979 13:`4 ignored content (same for any other digit)
- `12-25/1979` 13 ignored content (as the 13 is alone, we do not expect to be an hour)

Support for ISO-8601 will be great of course but concerning your argument of difficulty to test to avoid non-regression, it will be even harder after that PR since the rules seem more lenient.

Reply:
> Can you suggest an alternative way 

After a second look, I cannot find a way to improve readability without splitting into multiple line or dividing the process in multiple steps.

I have found 2 corner cases:
- 12-25/1979 :45:22 seems to match 45 as hour and 22 as minutes
- 12-25/1979 ::23 seems to match 23 as hour 
since hour / minute are mark as optional. Seems not the desired behavior but it's only a corner case I think.